controller.go

```diff
@@ -170,60 +172,78 @@ func (n *NodeController) nodeReady(node *v1.Node) (bool, error) {
 // setNodeReady sets node taint macthing ready value. E.g. sets NotReady taint
 // if ready is false, and removes the taint (if exists) when ready is true.
 func (n *NodeController) setNodeReady(node *v1.Node, ready bool) error {
-	// if ready, remove notReady taint if exists on the node
-	if ready {
-		var newTaints []v1.Taint
-		for _, taint := range node.Spec.Taints {
-			if taint.Key != n.taintNodeNotReadyName {
-				newTaints = append(newTaints, taint)
-			}
+	setNodeReadiness := func() error {
```

Comment:
Since there's already two cases to cover, I'd refactor this a bit. `setNodeReadiness` should be a simple function that gets and updates the state, and error-handling logic should in my opinion move into a separate wrapper.

Reply:
Refactored a bit, I'm completely sure on how to completely split it up because I would like to log only when there are changes. I'm open for ideas.

Reply:
The first thing I could think of is modifying `setNodeReadiness` to return (bool, error) with the semantics of either "old value, error" or "changed, error" and then logging/notifying hooks in the wrapper function. However, the current version looks a lot cleaner to me, so we can just leave it like this I guess.